src/oemof/solph/flows/_non_convex_flow_block.py

```diff
@@ -790,33 +782,28 @@ def _positive_gradient_flow_constraint(_):
         def _negative_gradient_flow_constraint(_):
             r"""Rule definition for negative gradient constraint."""
             for i, o in self.NEGATIVE_GRADIENT_FLOWS:
-                for index in range(1, len(m.TIMEINDEX) + 1):
-                    if m.TIMEINDEX[index][1] > 0:
+                for index in range(1, len(m.TIMESTEPS) + 1):
+                    if m.TIMESTEPS[index] > 0:
```

Comment:
Same here, why is there no iteration on the timesteps but over the index?

Reply:
Here, I do not see the point I made before. Maybe, the current implementation is faulty?

Reply:
I mean it works, it was just a side comment. I would say: merge the feature and maybe also make a release to ship the latest changes to the users.